src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -0,0 +1,112 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class SearchService {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final RestHighLevelClient client;
+
+    @Autowired
+    public SearchService(BigQueryPdao bigQueryPdao) {
+        // needs to be moved into own class with config
+        final RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200));
+        this.bigQueryPdao = bigQueryPdao;
+        this.client = new RestHighLevelClient(builder);
+    }
+
+    private String createEmptyIndex(Snapshot snapshot) {
+        String indexName = String.format("idx-%s", snapshot.getId());
+        try {
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest(indexName)
+                .settings(Settings.builder()
+                    .put("index.number_of_shards", 3));
+            CreateIndexResponse response = client.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+            if (!response.isAcknowledged()) {
+                throw new PdaoException("The index request was not acknowledged by one or more nodes");
+            }
+        } catch (IOException e) {
+            throw new PdaoException("Error creating index", e);
+        }
+        return indexName;
+    }
+
+    private void createIndexMapping(String indexName, List<Map<String, Object>> values) {
+        PutMappingRequest request = new PutMappingRequest(indexName);
+        Map<String, Object> properties = new HashMap<>();
+        values.get(0).forEach((key, val) -> {
+            properties.put(key, Map.of("type", "text"));
+        });
+        Map<String, Object> source = Map.of("properties", properties);
+        request.source(source);
+        try {
+            client.indices().putMapping(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            throw new PdaoException("Error creating index mapping", e);
+        }
+    }
+
+    private void addIndexData(String indexName, List<Map<String, Object>> values) {
+        BulkRequest request = new BulkRequest();
+        values.forEach(v -> {
+            request.add(new IndexRequest(indexName)
+                .id(v.get("uuid").toString())
+                .source(v, XContentType.JSON)
+            );
+        });
+        try {
+            if (!values.isEmpty()) {
+                client.bulk(request, RequestOptions.DEFAULT);
+            }
+        } catch (final IOException e) {
+            throw new PdaoException("Error indexing data", e);
+        }
+    }
+
+    private SearchIndexModel getIndexSummary(String indexName) {
+        GetIndexRequest request = new GetIndexRequest(indexName);
+        try {
+            GetIndexResponse getIndexResponse = client.indices().get(request, RequestOptions.DEFAULT);
+            SearchIndexModel searchIndexModel = new SearchIndexModel();
+            searchIndexModel.setIndexSummary(getIndexResponse.getIndices()[0]);
```

Comment:
This basically just returns the index id, which isn't ideal. I haven't found something equivalent to doing a `curl elasticsearch:9200/idx-<uuid>`